### PR TITLE
feature/connect-API

### PR DESCRIPTION
**Pull request by Kgomotso196**
# PROJECT TO DOs
### IN THIS PROJECT I:
1. Connected an existing API via Redux and used the data to display something in a UI.
2. Understood the concept of state management.
3. Used store, actions and reducers in React.
4. Connected React and Redux.
5. Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
6. Made sure that I used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
7. Made sure that I documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

### CODE REVIEW

File: src/components/BookList.js [hunk 1]
```diff
@@ -1,20 +1,54 @@
...
+
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const booksArr = useSelector((state) => state.book.books);
+
+  useEffect(() => {
+    const fetchBooksData = async () => {
+      try {
+        const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/d2UK603RcRIHbJQLulaw/books');
+        dispatch({ type: 'books/fetchBooks', payload: response.data });
```
Comment (brytebee) [inline src/components/BookList.js]: On page render or refresh, your store is not populated with data from your API. Kindly ensure that the store is properly updated. And the page is rendered with this data from the API.

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Kgomotso196 👋,

##

### Highlights💡
- Awesome use of `useEffect` and the dependency array.✔️
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookList.js (hunk 1 above)
Comment (Kgomotso196) [reply src/components/BookList.js]: @brytebee , thanks for your kind feedback, working on it.

**Review by roshan-bajgain — APPROVED** [review]
Hi @Kgomotso196, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Good job on implementing the changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**